docs/user-guide/lazy-loading.md

```diff
@@ -0,0 +1,88 @@
+# Lazy Loading
+
+> [!WARNING]
+> This is an experimental option and may not work as expected with all plugins.
+> The API may change without notice. > Please report any issues you encounter.
+
+When the lazy loading functionality is enabled, the plugin's configuration is
+routed to the enabled lazy provider. Options mentioned will be in relation to
+the `${namespace}.${name}.lazyLoad` context.
+
+## Supported Lazy Loading Providers
+
+It is recommended to become familiar with lazy loading strategies supported by
+your provider in their upstream documentation and the corresponding plugin's
+support.
+
+- [Lz.n](https://github.com/nvim-neorocks/lz.n?tab=readme-ov-file#plugin-spec)
+
+## Enabling Lazy Loading Per Plugin
+
+There are a couple different ways to enable lazy loading for a particular
+plugin.
+
+- `enable` set to `true`.
+- `settings` containing some configuration.
```

Comment:
```suggestion
- `lazyLoad.enable` set to `true`.
- `lazyLoad.settings` containing some configuration.
```
?

Reply:
Maybe we don't need to really mention `enable` here though; we could say something like 

> Configuring `lazyLoad.settings.*` will enable lazyloading. If needed, it can be explicitly disabled using `lazyLoad.enable = false`.

?

EDIT: I see you have something similar below
> If you are just wanting to store potential configuration without enabling it, you can explicitly disable it setting `enable = false`.

Reply:
I mention at the top that all the settings are in relation to the lazyLoad setting to reduce repetition everywhere. 
>When the lazy loading functionality is enabled, the plugin's configuration is
routed to the enabled lazy provider. Options mentioned will be in relation to
the `${namespace}.${name}.lazyLoad` context.

If this is easy to miss I can just inline it everywhere.

Reply:
> If this is easy to miss I can just inline it everywhere.

Yeah, I think `lazyLoad.settings` is clearer than `settings`, especially to anyone skim-reading.[^1] Having it mentioned explicitly at the top is great, but let's not rely _exclusively_ on that :upside_down_face: 

[^1]: Because we all _totally_ read third-party docs fully and in detail while rushing around trying to get our stuff to work :grin:

Reply:
While proofreading after the change, I do agree it is easier to follow.